static/app/components/events/interfaces/uptime/uptimeDataSection.tsx

```diff
@@ -30,40 +30,54 @@ const DOWNTIME_START_TYPES = [
 
 const DOWNTIME_TERMINAL_TYPES = [GroupActivityType.SET_RESOLVED];
 
-export function UptimeDataSection({group, event, project}: Props) {
-  const organization = useOrganization();
+export function useDowntimeDuration(group: Group): {
+  durationMs: number;
+  endDate: Date;
+  startDate: Date;
+} {
   const nowRef = useRef(new Date());
```

Comment:
i learned that this doesn't actually work as expected with rerenders

you have to either 
```tsx
const nowRef = useRef<Date>();
if (!nowRef.current) {
  nowRef.current = ...
}
```

or 

```
const [now] = useState(() => new Date())
```